### PR TITLE
CP-51325: Update Zip Sharp Library

### PR DESCRIPTION
**Pull request by ElijahSadorra**
- Update READ.me to correct new version of SharpZipLib, added clarifying steps and correct wrong command
- Update new version of SharpZipLib from 0.85.4 to 1.3.3 with 0 vulnerabilities
- Remove all SharpZipLib patches
- Update `.csproj` patch file to target net4.8

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -109,16 +109,24 @@ $gitCommit = git rev-parse HEAD
...
+Expand-Archive -DestinationPath "$SCRATCH_DIR\sharpziplib" -Path "$REPO\SharpZipLib\SharpZipLib-1.3.3.zip"	
 
-Get-ChildItem $PATCHES | where { $_.Name.StartsWith("patch-sharpziplib") } | % { $_.FullName } |`
-  applyPatch -Path "$SCRATCH_DIR\sharpziplib\SharpZipLib-1.3.3"
+Get-ChildItem $PATCHES | where { $_.Name.StartsWith("patch-sharpziplib") } | % { $_.FullName } |`	
+  applyPatch -Path "$SCRATCH_DIR\sharpziplib"	
 
-msbuild $SWITCHES $FRAME48 $VS_TOOLS $SIGN "$SCRATCH_DIR\sharpziplib\SharpZipLib-1.3.3\ICSharpCode.SharpZipLib.sln"
-'dll', 'pdb' | % { "$SCRATCH_DIR\sharpziplib\bin\ICSharpCode.SharpZipLib." + $_ } |`
+dotnet restore -v n "$SCRATCH_DIR\sharpziplib\SharpZipLib-1.3.3\src\ICSharpCode.SharpZipLib\ICSharpCode.SharpZipLib.csproj"
```
Comment (danilo-delbusso) [inline build.ps1:127]: since you're passing `$RESTORE_SWITCHES -p:RestoreConfigFile=$RESTORE_NUGET_CONFIG_FILE`, you don't need `dotnet restore`

**Review by danilo-delbusso — CHANGES_REQUESTED** [review]

**Comment by danilo-delbusso** [comment]
Sign off also missing from c878dc5beb05a5a767c7e371eec274829dbb10d2. Sorry, didn't see that earlier!

File: build.ps1 [hunk 2]
```diff
@@ -78,7 +78,7 @@ function applyPatch {
 
 $SWITCHES = '/nologo', '/m', '/verbosity:normal', '/p:Configuration=Release', `
             '/p:DebugSymbols=true', '/p:DebugType=pdbonly'
-$RESTORE_SWITCHES= '/restore', '/p:RestoreNoCache=true'
+$RESTORE_SWITCHES= '/restore', '/p:RestoreNoCache=true', "-p:RestoreConfigFile=$RESTORE_NUGET_CONFIG_FILE"
```
Comment (danilo-delbusso) [inline build.ps1:81]: For this to work you'll need to move line 114 above line 81

File: build.ps1 [hunk 3]
```diff
@@ -109,6 +109,25 @@ $gitCommit = git rev-parse HEAD
 git archive --format=zip -o "$OUTPUT_DIR\\dotnet-packages-sources.zip" $gitCommit
 "dotnet-packages.git $gitCommit" | Out-File -FilePath "$OUTPUT_DIR\dotnet-packages-manifest.txt"
 
+# prepare NuGet Config
+
+$RESTORE_NUGET_CONFIG_FILE="NuGet.Config"
+Copy-Item "$REPO\$RESTORE_NUGET_CONFIG_FILE" -Destination "$SCRATCH_DIR\$RESTORE_NUGET_CONFIG_FILE"
```
Comment (danilo-delbusso) [inline build.ps1:115]: here you've updated the contents of `Nuget.Config` in the scratch, but `RESTORE_NUGET_CONFIG_FILE` is still pointing to the old one

I think to keep things clean you should move the prepping of Nuget.Config before the initial assignment of RESTORE_SWTICHES, and make sure that the path we pass to `-p:RestoreConfigFile` is the one in the scratch directory.

**Review by danilo-delbusso — CHANGES_REQUESTED** [review]

File: build.ps1 (hunk 3 above)
Comment (danilo-delbusso) [reply build.ps1:115]: happy to have a chat if you need clarifications - DM anytime :D

File: patches/patch-sharpziplib-ca-61725.diff [hunk 4]
```diff
@@ -1,62 +0,0 @@
...
- 				longHeader.WriteHeader(this.blockBuffer);
- 				this.buffer.WriteBlock(this.blockBuffer);  // Add special long filename header block
- 
- 				int nameCharIndex = 0;
- 
--				while (nameCharIndex < entry.TarHeader.Name.Length) {
-+				while (nameCharIndex < byteCount) {
- 					Array.Clear(blockBuffer, 0, blockBuffer.Length);
--					TarHeader.GetAsciiBytes(entry.TarHeader.Name, nameCharIndex, this.blockBuffer, 0, TarBuffer.BlockSize);
-+					TarHeader.GetNameBytes(entry.TarHeader.Name, nameCharIndex, this.blockBuffer, 0, TarBuffer.BlockSize);
```
Comment (kc284) [inline patches/patch-sharpziplib-ca-61725.diff:59]: As discussed separately, there is a risk here:
- our patch uses longHeader.Size = byteCount;, while their code adds 1
- our patch uses TarHeader.GetNameBytes, while their code uses TarHeader.GetAsciiBytes passing in the additional parameter

File: patches/patch-sharpziplib-ca-63358-TarInputStream.diff [hunk 5]
```diff
@@ -1,316 +0,0 @@
-# Copyright (c) Cloud Software Group, Inc.
```
Comment (kc284) [inline patches/patch-sharpziplib-ca-63358-TarInputStream.diff:1]: Are we sure this patch is not needed anymore?

File: patches/patch-sharpziplib-ca-63358-TarBuffer.diff [hunk 6]
```diff
@@ -1,121 +0,0 @@
-# Copyright (c) Cloud Software Group, Inc.
```
Comment (kc284) [inline patches/patch-sharpziplib-ca-63358-TarBuffer.diff:1]: Are we sure this patch is not needed anymore?

**Review by kc284 — COMMENTED** [review]
I think we need to direct this PR first to a feature branch for testing and then to master, because there is risk associated with 3 out of the removed patches: the code in the new library has not been fixed (or has not been fixed in exactly the same way) and we don't have enough evidence yet that they're not needed.

**Review by kc284 — APPROVED** [review]
Approving so we can get a feature branch build to test.

**Review by danilo-delbusso — APPROVED** [review]

**Pull request merged**